### PR TITLE
Resolve documentation pointing to wrong SDK

### PR DESCRIPTION
**Pull request by marnovandermaas**
In the documentation we should just recommend that people use cmake to pull in the correct version of the SDK.

### CODE REVIEW

**Comment by marnovandermaas** [comment]
Partially fixes: https://github.com/lowRISC/sonata-system/issues/101

**Review by GregAC — APPROVED** [review]

File: doc/guide/building-examples.md [hunk 1]
```diff
@@ -27,56 +27,39 @@ You can drag and drop this UF2 file into the `SONATA` drive to program the firmw
...
 export CHERIOT_LLVM_BIN=/path/to/cheriot-llvm/bin
-export CHERIOT_RTOS_SDK=/path/to/cheriot-rtos/sdk
 ```
 
 > WARNING: The path to `/path/to/cheriot-llvm/bin` should point to the *build* directory you created, not just the root checked out `cheriot-llvm` directory.
 > The path will look something like: `~/llvm-tools/cheriot-llvm/builds/cheriot-llvm/bin`
 
+Please make sure you unset the `CHERIOT_RTOS_SDK` environment variable as `cmake` will automatically pull in [the correct version](https://github.com/lowRISC/cheriot-rtos/tree/sonata).
+
 The following assume you have run the `source .venv/bin/activate` command in the terminal you are using, and you are
```
Comment (HU90m) [inline doc/guide/building-examples.md:58]:
```suggestion
The following assumes you have run the `source .venv/bin/activate` command in the terminal you are using, and you are
```

File: doc/guide/building-examples.md [hunk 2]
```diff
@@ -101,3 +84,4 @@ You can load software onto the FPGA over USB (JTAG) using:
 ```
 There are actually four different ways of loading the program - we normally use JTAG for development, but you can also
 program it into the serial flash device on the board. See the page [Programming the Sonata Software](../dev/sw-programming.md).
+
```
Comment (HU90m) [inline doc/guide/building-examples.md:87]:
```suggestion
```

**Review by HU90m — CHANGES_REQUESTED** [review]

**Review by HU90m — COMMENTED** [review]
Also, there are a few sentences which are split over multiple lines.

**Comment by nwf** [comment]
I'm not sure I approve of increasingly relying on `cmake` to pull in the RTOS in a section for "advanced users".  Advanced users can probably be told how to ensure that they fetch the suitable version (which, hopefully soon, is just mainline?) and might want to make changes to the RTOS itself, which would be difficult if it's hidden behind `cmake`.  Thoughs?

**Comment by davidchisnall** [comment]
Why are you wrapping a build system that can target Sonata and comes with a dev container that just works in a load of CMake goo?

For reference, the first CHERIoT build system used CMake, but CMake was not sufficiently flexible to allow defining new kinds of targets and so we abandoned it.

**Comment by GregAC** [comment]
I would note the software in this repository is intended for system bring-up/test purposes. People wanting to actually develop software for Sonata will use https://github.com/lowRISC/sonata-software or indeed just the upstream CHERIoT RTOS repository: https://github.com/microsoft/cheriot-rtos/.

The main branch of this repository and of the https://github.com/lowRISC/sonata-software are our live development repositories and as such in a state of flux right now. So in particular documentation is still being worked on along with deciding exactly what lives where etc.

We do not intend for cmake to be the primary build system for Sonata software, rather we will use the existing build flows provided by the CHERIoT RTOS upstream. It may remain in use in this repository for the limited use case of building small programs used for verification purposes and only of relevance to hardware engineers working on the RTL but this is not yet decided.

File: doc/guide/building-examples.md (hunk 2 above)
Comment (marnovandermaas) [reply doc/guide/building-examples.md]: Thanks.

File: doc/guide/building-examples.md (hunk 1 above)
Comment (marnovandermaas) [reply doc/guide/building-examples.md]: Thanks

**Pull request merged**